elasticsearch_dsl/aggs.py

```diff
@@ -222,314 +1201,2502 @@ def to_dict(self) -> Dict[str, Any]:
 
+    :arg format: A numeric response formatter.
+    :arg value: Configuration for the field that provides the values.
+    :arg value_type:
+    :arg weight: Configuration for the field or script that provides the
+        weights.
+    """
 
-class SerialDiff(Pipeline[_R]):
-    name = "serial_diff"
+    name = "weighted_avg"
 
+    def __init__(
+        self,
+        *,
+        format: Union[str, "DefaultType"] = DEFAULT,
+        value: Union[
+            "types.WeightedAverageValue", Dict[str, Any], "DefaultType"
+        ] = DEFAULT,
+        value_type: Union[
+            Literal[
+                "string",
+                "long",
+                "double",
+                "number",
+                "date",
+                "date_nanos",
+                "ip",
+                "numeric",
+                "geo_point",
+                "boolean",
+            ],
+            "DefaultType",
+        ] = DEFAULT,
+        weight: Union[
+            "types.WeightedAverageValue", Dict[str, Any], "DefaultType"
+        ] = DEFAULT,
+        **kwargs: Any,
+    ):
+        super().__init__(
+            format=format, value=value, value_type=value_type, weight=weight, **kwargs
+        )
 
-class StatsBucket(Pipeline[_R]):
-    name = "stats_bucket"
 
+class VariableWidthHistogram(Bucket[_R]):
+    """
+    A multi-bucket aggregation similar to the histogram, except instead of
+    providing an interval to use as the width of each bucket, a target
+    number of buckets is provided.
+
+    :arg field: The name of the field.
+    :arg buckets: The target number of buckets. Defaults to `10` if
+        omitted.
+    :arg shard_size: The number of buckets that the coordinating node will
+        request from each shard. Defaults to `buckets * 50`.
+    :arg initial_buffer: Specifies the number of individual documents that
+        will be stored in memory on a shard before the initial bucketing
+        algorithm is run. Defaults to `min(10 * shard_size, 50000)`.
+    :arg script:
+    """
 
-class SumBucket(Pipeline[_R]):
-    name = "sum_bucket"
+    name = "variable_width_histogram"
 
+    def __init__(
+        self,
+        *,
+        field: Union[str, "InstrumentedField", "DefaultType"] = DEFAULT,
+        buckets: Union[int, "DefaultType"] = DEFAULT,
+        shard_size: Union[int, "DefaultType"] = DEFAULT,
+        initial_buffer: Union[int, "DefaultType"] = DEFAULT,
+        script: Union["types.Script", Dict[str, Any], "DefaultType"] = DEFAULT,
+        **kwargs: Any,
+    ):
+        super().__init__(
+            field=field,
+            buckets=buckets,
+            shard_size=shard_size,
+            initial_buffer=initial_buffer,
+            script=script,
+            **kwargs,
+        )
 
-class BucketSort(Pipeline[_R]):
-    name = "bucket_sort"
+    def result(self, search: "SearchBase[_R]", data: Any) -> AttrDict[Any]:
+        return FieldBucketData(self, search, data)
+
+
+class RandomSampler(Bucket[_R]):
```

Comment:
I'm manually adding the missing `RandomSampler` for now. I think once it is added to the schema this is going to cause a failure in the type checking, so I'll know I can remove it then.

Reply:
`RandomSampler` is in the specification now. I've just backported to 8.15 in case it helps here.